### PR TITLE
chore: update tech stack of backend

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
fix https://github.com/fluidemocracy/backend-legacy/issues/1
/claim https://github.com/fluidemocracy/backend-legacy/issues/1

### CODE REVIEW

**Comment by leog** [comment]
@rishi-raj-jain thanks for creating this

I understand that this solves updating the tech stack of the backend, but there is no way to make sure it covers all the bases for the frontend to be certain this PR effectively solves what it should, so I'm reluctant to approve this PR without being sure the frontend works as expected using the new model.

File: prisma/schema.prisma [hunk 1]
```diff
@@ -2,69 +2,106 @@
...
-    title    String
-    body     String
-    category PostCategory
+model Session {
+  id           String   @id @default(cuid())
+  sessionToken String   @unique
+  userId       String
+  expires      DateTime
+  user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
+}
```
Comment (leog) [inline prisma/schema.prisma:41]: Session and Account should be handled by `next-auth` so it shouldn't really be a straight conversion from legacy

File: src/lib/prisma.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
```
Comment (yashug) [inline src/lib/prisma.ts:3]: @rishi-raj-jain this approach creates lot of prisma instances in dev mode when hot reload happens. i suggest this approach here 

```
import { PrismaClient } from "@prisma/client";

declare global {
  var prisma: PrismaClient | undefined;
}

export const db = globalThis.prisma || new PrismaClient();

if (process.env.NODE_ENV !== "production") globalThis.prisma = db;
```

**Comment by rishi-raj-jain** [comment]
@leog Can you authorize the latest deploy?

**Comment by rishi-raj-jain** [comment]
> so I'm reluctant to approve this PR without being sure the frontend works as expected using the new model.

Makes sense, I'm gonna run the docker version of things you mentioned that is now working to see how the frontend looks. Will update the thread here.

**Comment by rishi-raj-jain** [comment]
@leog

So auth does require it's own setup in Prisma when using Prisma Adapter (as in the template already). Tested and pushed a fix.

The command to run the project now:

```
brew services start postgresql@15
pnpm i
npx prisma generate
npx prisma migrate dev
npx prisma db seed
pnpm dev
```

**Comment by leog** [comment]
@rishi-raj-jain I'm cancelling this work, sorry for the inconveniences

**Comment by rishi-raj-jain** [comment]
I'm ready to join.

**Comment by rishi-raj-jain** [comment]
@leog 

I'm sorry to be outrightly asking for it, but may you atleast tip 30 - 50 dollars for the PR I did here?

**Comment by rishi-raj-jain** [comment]
@leog 

Are you joining the meet?

**Comment by leog** [comment]
sure @rishi-raj-jain, let me know your paypal account

**Comment by rishi-raj-jain** [comment]
@leog 

rishi18304@iiitd.ac.in

**Pull request closed without merging**